README.md

```diff
@@ -456,6 +456,12 @@ ANSWERS.addComponent('SearchBar', {
     enabled: false,
     // Optional, the message in the alert. Defaults to the below
     message: "We are unable to determine your location"
+  },
+  // Optional, functions invoked when certain events occur
+  customHooks: {
+    // Optional, a function invoked when a search is submitted via this component. The users search
+    // is passed in as a string
+    onSubmit: function(searchTerms) {}
```

Comment:
This hook is only fired when `onQuerySubmit` is fired. Can a query be submitted in this component without going through `onQuerySubmit`? If so, we should make explicit the cases that `onSubmit` will be fired.

Reply:
In a case where the global storage's "StorageKeys.QUERY" is set outside of this function, a search is conducted and the `onSubmit` wouldn't be fired.

However, since that case involves a search as a side effect from a separate action, I think the concept of "that search was not submitted via this component" makes sense. The argument the other way is: technically the "debouncedSearch" function lives in this component/file; however, that seems like an implementation detail. What do you think?

Reply:
I think it makes sense to only fire the hook when a search happens from within this component

Reply:
Sure, the distinction makes sense to me!